source/fab/steps/c_pragma_injector.py

```diff
@@ -0,0 +1,38 @@
+"""
+Add custom pragmas to C code which identify user and system include regions.
+
+"""
+from pathlib import Path
+from typing import Dict
+
+from fab.steps import Step
+from fab.tasks.c import _CTextReaderPragmas
+from fab.util import SourceGetter, FilterFpaths
+
+DEFAULT_SOURCE_GETTER = FilterFpaths('all_source', ['.c'])
+
+
+# todo: test
+class CPragmaInjector(Step):
```

Comment:
(assuming you meant _"part of doing the pre-processing"_) We too merged the pragma injection with the c pre-processing at one point. Because the fortran and c pre-processing are almost identical, even using the same exe, we got cleaner code by splitting it out (removing special cases). Another benefit is that we can demonstrate/explore the flexibility of how data flows between steps in the UM config. [Here we can see](https://github.com/bblay/fab/blob/4cd4c2d2e08ee8ddfdb010b9413050ca256c1af7/run_configs/um/build_um.py#L128-L161) the fortran preprocessor needs no instructions on what to process, but the c preprocessor includes the [optional config](https://github.com/bblay/fab/blob/4cd4c2d2e08ee8ddfdb010b9413050ca256c1af7/run_configs/um/build_um.py#L131) telling it to process the output from the pragma injector.

Of course we can always put it into the pre-processing if required, too.